### ISSUE TITLE
Update appRoleAssignment example

### ISSUE THREAD

**Issue by rikkuness** [microsoftgraph/microsoft-graph-docs#5869]
The example request on this page shows a `PATCH https://graph.microsoft.com/beta/appRoleAssignments/{id}` which seems to be contradictory of the 3 URL examples further up the page. If you attempt a `PATCH` to the URI shown in the example you get a 400 with `"Direct queries to this resource type are not supported."` in the body.

Using another of the examples (`PATCH /servicePrincipals/{id}/appRoleAssignedTo`) returns an error 405 with `"Specified HTTP method is not allowed for the request target."`

I've also attempted to use another example (`PATCH /users/{id | userPrincipalName}/appRoleAssignments/{id}`) but that example returns a 400 with `"Invalid resource identifier for EntitlementGrant."`.

I'm not certain what things are documentation errors and what are API bugs, any which way I attempt to update a users app role assignments fails.

Thank you!

UPDATE:  It seems like a `POST /servicePrincipals/{id}/appRoleAssignedTo` where `id` is the application ID seems to work as expected, so perhaps this is a documentation issue and `PATCH` should be `POST`? It's unclear if the request body should differ for a `POST`, I didn't seem to lose any existing assignments when doing a `POST`.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 81066b37-6e22-41d8-5b60-d1f3dc692c77
* Version Independent ID: 837a114f-9044-76de-f6fb-498573776044
* Content: [Update approleassignment - Microsoft Graph beta](https://docs.microsoft.com/en-us/graph/api/approleassignment-update?view=graph-rest-beta&tabs=http)
* Content Source: [api-reference/beta/api/approleassignment-update.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/beta/api/approleassignment-update.md)
* Product: ****
* Technology: **microsoft-graph**
* GitHub Login: @jthake-msft
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by aronfelipe** [comment]
How to get this work? I am getting 400 "Not a valid reference update." on POST /servicePrincipals/{id}/appRoleAssignedTo

**Comment by devgeeklav** [comment]
Documentation is incorrect, none of the HTTP requests work
 PATCH /users/{id | userPrincipalName}/appRoleAssignments/{id} 

Error:
"error": {
        "code": "Request_UnsupportedQuery",
        "message": "Invalid resource identifier for EntitlementGrant.",
        "innerError": {
            "request-id": "74d9faff-f81c-476e-af48-2559cc0b6631",
            "date": "2020-02-19T18:46:25"
        }
    }
}



PATCH /servicePrincipals/{id}/appRoleAssignedTo 


{
    "error": {
        "code": "Request_BadRequest",
        "message": "Specified HTTP method is not allowed for the request target.",
        "innerError": {
            "request-id": "a9c4dd7f-88b9-413e-baae-88144efb0798",
            "date": "2020-02-19T18:51:58"
        }
    }
}

None of the requests with PATCH works, i even tried POST request...the role assignment always gives 400 error.

**Comment by lozog95** [comment]
Hi! 
Using trial and error method I was able finally to do some calls which assigned a group to the application. I agree that documentation has some mistake (i used POST) and also the response didn't stand for a successful call. Please take a look:

**Request**: POST https://graph.microsoft.com/beta/servicePrincipals/<APP_ID>/appRoleAssignedTo
Body: 
```
{
  "principalDisplayName": "<Group display name>",
  "principalId": "<Group ID>",
  "resourceId": "<APP_ID>",
  "principalType": "Group",
  "resourceDisplayName": "<Application name>"
}
```

> pls note: **APP_ID** is not the **Application ID** (as in the app registrations) but it's Object ID (also seemed a bit confusing to me).

**Response**:
Status code: 500 :)
```
{
    "error": {
        "code": "InternalServerError",
        "message": "Error while processing response.",
        "innerError": {
            "request-id": "<someid>",
            "date": "2020-04-07T15:50:24"
        }
    }
}

```
However, despite the fact I received 500 and this scary message, the group was added to the Enterprise application.

**Comment by brownm5** [comment]
I have tried
POST https://graph.microsoft.com/beta/servicePrincipals/e19b155f-b3dd-4eb4-a984-ab3bfadbd570/appRoleAssignedTo

And received Status code 400 - "Not a valid reference update."

I have also tried (the guid is the application objectId)
PATCH https://graph.microsoft.com/beta/appRoleAssignments/e19b155f-b3dd-4eb4-a984-ab3bfadbd570
and received
Status 400 - "Direct queries to this resource type are not supported."

and tried this (the guid is the application objectId)  
POST https://graph.microsoft.com/beta/appRoleAssignments/e19b155f-b3dd-4eb4-a984-ab3bfadbd570
And received
Status 405 "Specified HTTP method is not allowed for the request target."


Using the same body on all attempts
{
  "creationTimestamp": "2016-10-19T10:37:00Z",
  "principalDisplayName": "awsrole-app01d1-support",
  "principalId": "ffc58aa8-37e5-4bbe-a3f3-4c419cbeb3b9",
  "principalType": "Group",
  "resourceDisplayName": "Amazon Web Services (AWS)"
}


Does anyone have updated documentation....  or Am I doing something wrong??
Any help would be appreciated

**Comment by devgeeklav** [comment]
After different permutation and combination, worked for me, here are the details:
API End Point: 
https://graph.microsoft.com/beta/servicePrincipals/<ObjectGUID of Service Principal>/appRoleAssignedTo

Note that ObjectGUID is the the objectID of App from Enterprise Application

HTTP Request: PATCH

Sample JSON Request in PATCH Call:
{
"appRoleId": "<objectGUID of AppRole>",
"principalId": "ObjectGUID of Group",
"principalType": "Group",
"resourceDisplayName": "DisplayName of App/Resource",
"resourceId": "ObjectGUID of ServicePrincipal"
}

**Comment by psignoret** [comment]
Thank you everyone for you comments and feedback (and please accept my apologies for the lack of response here). 

The documentation for app role assignments has now been updated, and the API has also now been released to v1.0.

The short version:
* To assign an app role to a user, group or service principal,[`POST to /v1.0/servicePrincipals/{resource-id}/appRoleAssignedTo`](https://docs.microsoft.com/graph/api/serviceprincipal-post-approleassignedto?view=graph-rest-1.0&tabs=http), where `{resource-id}` is the **id** of the resource app's  **servicePrincipal** object. In the body, include only `principalId` (the **id** of the user, group or service principal to which the app role is being granted), `resourceId` (same as `{resource-id}`), and `appRoleId` (the **id** of the **appRole** to grant, or 0-Guid if there are none).
* To remove an app role assignment from a user, group or service principal, do a [`DELETE to /v1.0/servicePrincipals/{resource-id}/appRoleAssignedTo/{assignment-id}`](https://docs.microsoft.com/graph/api/serviceprincipal-delete-approleassignedto?view=graph-rest-1.0&tabs=http), where `{resource-id}` is the **id** of the resource app's  **servicePrincipal** object and `{assignment-id}` is the **id** of the specific **appRoleAssignment** you want to delete. (You can search for app role assignments (e.g. to find the `{assignment-id}` by queries (with filters) on either the **appRoleAssignments** navigation on the assigned user, group or service principal, or on the **appRoleAssignedTo** navigation on the resource app's service principal.)

I'm going to close this issue because it's likely outdated given the update to documentation, but please feel free to open a new one (or add to/re-open this one, if that's an option for you) if you're still having issues.
